src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,134 @@
+package duke.storage;
+
+import duke.Duke;
+import duke.command.Parser;
+import duke.exception.DukeParserException;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TodoTask;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Represents the storage manager for <code>Duke</code>.
+ * Reads and writes the task list to file.
+ *
+ * @author Terence Chong Guang Jun
+ */
+public class Storage {
+    private final Charset charset;
+    private final Path filePath;
+
+    /**
+     * Creates a new storage with the specified file path in UTF-8 encoding.
+     *
+     * @param path the path of the file for reading and writing.
+     */
+    public Storage(String path) {
+        this(path, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Creates a new storage with the specified file path and character encoding.
+     *
+     * @param path the path of the file for reading and writing.
+     * @param charset the character encoding to be used.
+     */
+    public Storage(String path, Charset charset) {
+        filePath = Path.of(path).normalize();
```

Comment:
Could it be just fine with only one constructor?
Pardon me because I personally think with just one will work perfectly fine

Reply:
In this case I'm just handling a case where an implementer might want to use a different encoding than UTF-8 haha